commons/src/main/java/org/infinispan/commons/marshall/WrappedByteArray.java

```diff
@@ -125,4 +126,30 @@ public WrappedByteArray readObject(ObjectInput input) throws IOException {
          }
       }
    }
+
+   public static class Marshaller implements MessageMarshaller<WrappedByteArray> {
+
+      @Override
+      public Class<? extends WrappedByteArray> getJavaClass() {
+         return WrappedByteArray.class;
+      }
+
+      @Override
+      public String getTypeName() {
+         return "persistence.WrappedByteArray";
+      }
+
+      @Override
+      public WrappedByteArray readFrom(ProtoStreamReader reader) throws IOException {
+         byte[] bytes = reader.readBytes("wrappedBytes");
+         int hashcode = reader.readInt("hashcode");
```

Comment:
why marshall haschode? cant we recompute it? or let it be recomputed lazily

Reply:
This was added a while back to optimize the case with very large key or values.

Reply:
I guess we should probably remove it from this marshaller then as I imagine it will not make a significant difference when loading from persistence.